### PR TITLE
Dynamic Contact Information and History

### PR DESCRIPTION
**Pull request by VishalZ123**
## Dynamic Contact Info and History

**Summary**
Implements the dynamic contact info and history. Uses `GET_CONTACT_INFO` for fetching the contact information, formats it and displays. More info option displays the fields for the contact. Contact History option displays all the action taken for the contact, using the `GET_CONTACT_HISTORY` query.

**Changes Made**
- Implements dynamic contact information
- Adds More info page that displays the fields for the contacts
- Adds the Contact History page that displays the history actions of a contact
- Load more history on scroll
- Adds tests and mocks for unit testing

https://github.com/glific/mobile/assets/76409986/4f44ae6f-2b8c-4de4-b372-50a3e298ec22

**Issues**

Closes: #140 

**Reviewers**
@mdshamoon , Kindly review and merge

### CODE REVIEW

**Comment by ajaman190** [comment]
Hey @VishalZ123, please resolve all the deep scan issues.

**Review by ajaman190 — CHANGES_REQUESTED** [review]

File: screens/ContactInfo.tsx [hunk 1]
```diff
@@ -1,19 +1,30 @@
...
       <View style={styles.bodyContainer}>
-        <FieldValue field={'Provider status'} value={'Can send template messages'} />
-        <FieldValue field={'Status'} value={'Optin via WA on 22/03/2023, 11:59:57'} />
-        <FieldValue field={'Role'} value={'Student'} />
-        <FieldValue field={'Age'} value={'11 to 14'} />
-        <FieldValue field={'Language'} value={'English'} />
-        <FieldValue testID={'contactStatus'} field={'Status'} value={'Valid contact'} />
+        {Object.keys(fields).length === 0 ? (
+          <Text style={styles.placeholder}>No Fields Available</Text>
+        ) : (
```
[the remark this answers is not in the document]
Comment (VishalZ123) [reply screens/ContactInfo.tsx:23]: If there are no fields then the fields are empty, like not even title are present.
These are created as key value pairs while creating the flows.

File: __tests__/ContactProfile.test.tsx [hunk 2]
```diff
@@ -43,11 +44,39 @@ describe('ContactProfile', () => {
...
+        fields: { Name: 'John', Age: '15 to 18' },
+      });
+    });
+  });
 
+  test('navigate to contact history page', async () => {
+    const navigationMock = {
+      navigate: jest.fn(),
+      goBack: jest.fn(),
+    };
```
[the remark this answers is not in the document]
Comment (VishalZ123) [reply __tests__/ContactProfile.test.tsx:70]: okay, I must have overlooked the use of `navigationMock` in multiple places in the same fiel

File: __tests__/ContactHistory.test.tsx [hunk 3]
```diff
@@ -2,14 +2,67 @@ import React from 'react';
...
 
-    expect(getByText('Load More')).toBeDefined();
+    const { getByText, getByTestId } = customRender(
+      <ContactHistory route={{ params: { id: '2' } }} />,
+      GET_CONTACT_HISTORY_MOCK
+    );
+    await waitFor(() => getByTestId('historyFlatList'));
+    const flatList = getByTestId('historyFlatList');
+    // fire the event to fetch more history
+    fireEvent(flatList, 'onEndReached');
```
[the remark this answers is not in the document]
Comment (VishalZ123) [reply __tests__/ContactHistory.test.tsx:61]: I don't get it, the line which you changed was not using the `eventData`, can you clarify
also, can you please specify the test in chat screen which you are talking about. The file name and line. there's no test as "renders correctly" in `ChatScreen.test.tsx`

**Review by mdshamoon — APPROVED** [review]

**Pull request merged**